bosun-monitor/bosun
# Add ElasticSearch 7.x support

# Description
This pull request adds ES 7.x support in Bosun.
I submitted https://github.com/bosun-monitor/bosun/pull/2427 before but that is based on my own go modules migration pull request which is different from what upstream maintainers did. This one is based on upstream go modules migration.

- https://github.com/olivere/elastic/ version 7 client is used.
- Most code are copy-pasted from ES version 6 support.
- The one difference is in ES 7 [mapping types is removed](https://www.elastic.co/guide/en/elasticsearch/reference/master/removal-of-types.html), [Type(docType)](https://github.com/bosun-monitor/bosun/blob/f4392c2a8852b753e7490c27a599c04876c912c7/annotate/backend/elastic6.go#L161) are removed in ES7

## Type of change

From the following, please check the options that are relevant.


# How has this been tested?

Tested with ES 7.2 and also ES 6.8 to make sure nothing breaks.
- update the configuration toml file and create some alerts to query ES
- verify alerts can be triggered and no errors in ES query
- verify annotate can work

# Checklist:

- [x] This contribution follows the project's [code of conduct](https://github.com/bosun-monitor/bosun/blob/master/CODE_OF_CONDUCT.md)
- [x] This contribution follows the project's [contributing guidelines](https://github.com/bosun-monitor/bosun/blob/master/CONTRIBUTING.md)
